accounts-db/src/accounts_db.rs

```diff
@@ -1779,10 +1782,16 @@ impl AccountsDb {
     const DEFAULT_MAX_READ_ONLY_CACHE_DATA_SIZE_HI: usize = 410 * 1024 * 1024;
 
     pub fn default_for_tests() -> Self {
-        Self::default_with_accounts_index(AccountInfoAccountsIndex::default_for_tests(), None, None)
+        Self::default_with_accounts_index(
+            &ACCOUNTS_DB_CONFIG_FOR_TESTING,
+            AccountInfoAccountsIndex::default_for_tests(),
+            None,
+            None,
+        )
     }
 
     fn default_with_accounts_index(
+        accounts_db_config: &AccountsDbConfig,
```

Comment:
This is how I got around creating the thread pool twice. With my other refactor in https://github.com/anza-xyz/agave/pull/3270, I think `default_with_accounts_index()` could go away and just become part of `new_with_config()`. This could seemingly be more cleanup outside of this PR; happy to continue down this route of cleanup